migrations/v2/V202408201553__v2.4.x__ContentAutotagging.sql

```diff
@@ -318,15 +321,1513 @@ EXEC [${flyway:defaultSchema}].spUpdateEntityFieldRelatedEntityNameFieldMap @Ent
+INSERT INTO [${flyway:defaultSchema}].ContentFileType ( Name, FileExtension ) VALUES ( 'XML', '.xml' )
+INSERT INTO [${flyway:defaultSchema}].ContentFileType ( Name, FileExtension ) VALUES ( 'HTML', '.html' )
+
+INSERT INTO [${flyway:defaultSchema}].ContentSourceType ( Name, Description ) VALUES ( 'Local File System', NULL )
+INSERT INTO [${flyway:defaultSchema}].ContentSourceType ( Name, Description ) VALUES ( 'Website', NULL )
+INSERT INTO [${flyway:defaultSchema}].ContentSourceType ( Name, Description ) VALUES ( 'RSS Feed', NULL )
+INSERT INTO [${flyway:defaultSchema}].ContentSourceType ( Name, Description ) VALUES ( 'Cloud Storage', NULL )
+
+INSERT INTO [${flyway:defaultSchema}].ContentSourceTypeParam (Name, Description, Type, DefaultValue, IsRequired ) VALUES ('CrawlOtherSitesInTopLevelDomain', 'Dictates whether we crawl links from a website at the same domain level as the link provided.', 'Boolean', 'false', 1)
+INSERT INTO [${flyway:defaultSchema}].ContentSourceTypeParam (Name, Description, Type, DefaultValue, IsRequired ) VALUES ('CrawlSitesInLowerLevelDomain', 'Dictates whether we crawl sites that are in lower level domains from the starting URL.', 'Boolean', 'false', 1)
+INSERT INTO [${flyway:defaultSchema}].ContentSourceTypeParam (Name, Description, Type, DefaultValue, IsRequired ) VALUES ('MaxDepth', 'Specifies the maximum depth that we crawl the site.', 'Number', '0', 1)
+INSERT INTO [${flyway:defaultSchema}].ContentSourceTypeParam (Name, Description, Type, DefaultValue, IsRequired ) VALUES ('RootURL', 'Optional parameter to specify what form any scraped links must start with, and defaults to the homepage of the website.', 'String', Null, 0)
+INSERT INTO [${flyway:defaultSchema}].ContentSourceTypeParam (Name, Description, Type, DefaultValue, IsRequired ) VALUES ('URLPattern', 'Optional parameter for the Webcrawler that defines a regular expression pattern that the URL must match when extracting links on a page. The default is a regular expression that allows for any link.', 'RegExp', '^.*$', 0)
+
+DROP PROCEDURE IF EXISTS [__mj].[spCreateContentFileType]
+GO
+
+CREATE PROCEDURE [__mj].[spCreateContentFileType]
+    @Name nvarchar(255),
+    @FileExtension nvarchar(255)
+AS
+BEGIN
+    SET NOCOUNT ON;
+    DECLARE @InsertedRow TABLE ([ID] UNIQUEIDENTIFIER)
+    INSERT INTO 
+    [__mj].[ContentFileType]
+        (
+            [Name],
+            [FileExtension],
+            [__mj_CreatedAt],
+            [__mj_UpdatedAt]
+        )
+    OUTPUT INSERTED.[ID] INTO @InsertedRow
+    VALUES
+        (
+            @Name,
+            @FileExtension, 
+            GETUTCDATE(),
+            GETUTCDATE()
+        )
+    -- return the new record from the base view, which might have some calculated fields
+    SELECT * FROM [__mj].[vwContentFileTypes] WHERE [ID] = (SELECT [ID] FROM @InsertedRow)
+END
+GO
+GRANT EXECUTE ON [__mj].[spCreateContentFileType] TO [cdp_Developer], [cdp_Integration]
```

Comment:
Not sure if it's coming up, but don't forget that you will also need to INSERT EntityPermission rows to grant privs on the ContentFileType entity for users